sos/report/plugins/ceph_osd.py

```diff
@@ -39,9 +39,62 @@ def setup(self):
 
         self.add_cmd_output([
             "ceph-disk list",
-            "ceph-volume lvm list",
+            "ceph-volume lvm list"
         ])
 
+        ceph_cmds = [
+           "bluestore bluefs available",
+           "config diff",
+           "config show",
+           "dump_blacklist",
+           "dump_blocked_ops",
+           "dump_historic_ops_by_duration",
+           "dump_historic_slow_ops",
+           "dump_mempools",
+           "dump_ops_in_flight",
+           "dump_op_pq_state",
+           "dump_osd_network",
+           "dump_reservations",
+           "dump_watchers",
+           "log dump",
+           "perf dump",
+           "perf histogram dump",
+           "objecter_requests",
+           "ops",
+           "status",
+           "version",
+        ]
+
+        osd_ids = []
+        out = self.exec_cmd('ps -u ceph -o args')
+
+        if out['status'] != 0:
+            return
+
+        # Get the running OSD process IDs from ps
+        osd_nums = [
+                m.split()[5] for m in out['output'].splitlines()
```

Comment:
Are we sure each line will have at least 6 words, and no `IndexError` exception can be raised?

Reply:
Yes I think we can expect this output to be consistent like the format I see today, but I will check again with @pponnuvel or @hillpd if this might not be the case. I'd expect the id number always to be element number 6.

root@bronzor:~/sos/sos# ps -u ceph -o args
COMMAND
/usr/bin/ceph-osd -f --cluster ceph --id 0 --setuser ceph --setgroup ceph
/usr/bin/ceph-osd -f --cluster ceph --id 1 --setuser ceph --setgroup ceph
/usr/bin/ceph-osd -f --cluster ceph --id 2 --setuser ceph --setgroup ceph

Reply:
It may be possible that there are other ceph processes running on the same host, not just OSDs. So it's possible to hit `IndexError`.

This could be handled by checking:
- the line contains at least 6 fields
- 5th field is `--id`
- 6th field is a number (not a string e.g. mon which could be running on the same machine in a hyperconverged env)

Which would rule out others except OSDs.

Reply:
ack @pponnuvel , done